safe_transaction_service/history/services/index_service.py

```diff
@@ -162,31 +162,33 @@ def txs_create_or_update_from_tx_hashes(
                 raise TransactionNotFoundException(
                     f"Cannot find tx with tx-hash={HexBytes(tx_hash).hex()}"
                 )
-            elif tx.get("blockNumber") is None:
+            elif tx.get("blockHash") is None:
                 raise TransactionWithoutBlockException(
-                    f"Cannot find blockNumber for tx with "
+                    f"Cannot find blockHash for tx with "
                     f"tx-hash={HexBytes(tx_hash).hex()}"
                 )
-            block_numbers.add(tx["blockNumber"])
+            block_hashes.add(tx["blockHash"].hex())
             txs.append(tx)
 
-        blocks = self.ethereum_client.get_blocks(block_numbers)
+        blocks = self.ethereum_client.get_blocks(block_hashes)
         block_dict = {}
-        for block_number, block in zip(block_numbers, blocks):
+        for block_hash, block in zip(block_hashes, blocks):
             block = block or self.ethereum_client.get_block(
-                block_number
+                block_hash
             )  # Retry fetching if failed
             if not block:
                 raise BlockNotFoundException(
-                    f"Block with number={block_number} was not found"
+                    f"Block with hash={block_hash} was not found"
                 )
-            assert block_number == block["number"]
-            block_dict[block["number"]] = block
+            assert (
+                block_hash == block["hash"].hex()
+            ), f'{block_hash} - {block["hash"].hex()}'
+            block_dict[block["hash"]] = block
```

Comment:
Here the logic changed a bit (I mean the substraction). Why this operation necessary with `block_hash` but not `block_number`?
Also, is this `=` assignment atomic? Could this be an issue in a concurrent scenario?

Reply:
Oh, that's a debug line I added. I've just removed it. Yes, it's atomic